pkg/driver/agent.go

```diff
@@ -79,11 +87,86 @@ func NewNode(d *CSIDriver) csi.NodeServer {
 		}
 	}()
 
+	if d.config.ListenAddress != "" {
+		exposeMetrics(d.config.ListenAddress, d.config.MetricsPath, d.config.DisableExporterMetrics)
+	}
+
 	return &node{
 		driver: d,
 	}
 }
 
+/*
+Function to register collectors to collect LVM related metrics and exporter metrics.
+
+If disableExporterMetrics is set to false, exporter will include metrics about itself i.e (process_*, go_*).
+*/
+func registerCollectors(disableExporterMetrics bool) (*prometheus.Registry, error) {
+	registry := prometheus.NewRegistry()
+
+	if !disableExporterMetrics {
+		processCollector := collectors.NewProcessCollector(collectors.ProcessCollectorOpts{})
+		err := registry.Register(processCollector)
+		if err != nil {
+			klog.Errorf("failed to register process collector for exporter metrics collection: %s", err.Error())
+			return nil, err
+		}
+		goProcessCollector := collectors.NewGoCollector()
+		err = registry.Register(goProcessCollector)
+		if err != nil {
+			klog.Errorf("failed to register go process collector for exporter metrics collection: %s", err.Error())
+			return nil, err
+		}
+	}
+	lvmCollector := collector.NewLvmCollector()
+
+	err := registry.Register(lvmCollector)
+	if err != nil {
+		klog.Errorf("failed to register LVM collector for LVM metrics collection: %s", err.Error())
+		return nil, err
+	}
+	return registry, nil
+}
+
+/*
+Function to start HTTP server to expose LVM metrics.
```

Comment:
Commenting format needs to be fixed. No new lines between each sentences. Use `//` for public function comments etc. See [this](https://golang.org/doc/effective_go#commentary) from official effective go guidelines.